clar2wasm/tests/wasm-generation/principal.rs

```diff
@@ -184,4 +206,26 @@ mod clarity_v2_v3 {
             );
         }
     }
+
+    proptest! {
+        #![proptest_config(runtime_config())]
+
+        #[test]
+        fn crosscheck_principal_of(
+            key in buffer(33)
+        ) {
+            let hex_string = &format!("{}", key)[2..];
+            let pubkey = Secp256k1PublicKey::from_hex(hex_string);
+
+            let (snippet, expected) = match pubkey.is_ok() {
+                true => (&format!("(is-standard (try! (principal-of? {key})))"),  Value::Bool(true)),
```

Comment:
If `principal-of?` were to return a wrong (but valid) principal for the public key, this check would still pass

Reply:
Changed the implementation.